### PR TITLE
Ocean/sediment transport design doc

### PR DESCRIPTION
**Pull request by caozd999**
This PR adds the prognostic sediment transport design document. It would also be used for discussions about where and how
-  to define the sediment transport related variables (have been/will be listed in the design document) 
-  to implement each development requirement  

This design document will keep updated along with the code development work.

### CODE REVIEW

**Comment by xylar** [comment]
@caozd999, I've added @mark-petersen and myself as reviewers but neither of us knows anything about sediment transport.  Who else should review for the science side, not just technical details?

**Comment by xylar** [comment]
@caozd999, thanks for submitting this at my request!  I'm going to do a first pass through the design doc just for formatting issues related to RestructuredTest and sphinx.  I hope that's okay.  I probably won't get to that until tomorrow, though.  Too many meetings today.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:27]: If you want this to come out formatted like a numbered list, you need blank lines:
```suggestion

1) sediment classes;

2) bedload sediment transport;

3) suspended sediment transport; and

4) morphological change.

```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:97]:
```suggestion
- `Zyserman and Fredsoe, 1994 <https://ascelibrary.org/doi/pdf/10.1061/%28ASCE%290733-9429%281994%29120%3A9%281021%29>`_

- `Goldstein et al., 2014 <https://esurf.copernicus.org/articles/2/67/2014/esurf-2-67-2014-discussion.html>`_
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:109]:
```suggestion
- `Soulsby, 1997 <https://www.icevirtuallibrary.com/doi/abs/10.1680/doms.25844.fm>`_

- `Cheng, 1997 <https://doi.org/10.1061/(ASCE)0733-9429(1997)123:2(149)>`_
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:160]:
```suggestion
.. math::

	\theta = \frac{\tau}{g(s-1)D_{50}}
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:149]:
```suggestion
.. math::

    q_b = \Phi \sqrt{(s-1)g D_{50}^3}\rho_s
```

Comment (xylar) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:149]: Same in many other places.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:352]:
```suggestion
  And there are three options for it:

        * :code:`Engelund-Hansen`

	* :code:`Meyer-Peter-Mueller`

	* :code:`Soulsby-Damgaard`
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:160]: You're also using tabs for indenting in a few places.  The document should not have any tabs, just spaces.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:378]:
```suggestion

- Pre-defined sediment-related variables

  * sediment grain diameter: :code:`sedimentGrainDiameter(nSedimentClasses, nCells)`

  * sediment grain density: :code:`sedimentGrainDensity(nSedimentClasses, nCells)`

  * sediment class fraction on bed layer: :code:`sedimentClassFraction(nSedimentClasses)`

  * bedload transport coefficient: :code:`sedimentBedloadCoefficient(nSedimentClasses)`

- Other variables

  * bedload transport rate: :code:`sedimentBedloadTransportRate(nCells)`
```

Comment (xylar) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:378]: I guess you get the idea.  Many of these lists need this kind of spacing.

**Comment by xylar** [comment]
@caozd999, I think that's all my tech feedback for now.

**Comment by caozd999** [comment]
> @caozd999, I've added @mark-petersen and myself as reviewers but neither of us knows anything about sediment transport. Who else should review for the science side, not just technical details?

I think LeAnn should be added here -- she is team member of this task.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:165]: Thanks. Is there any specific number of spaces I should place for the indenting format?

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:154]: I will work through the document to fix them.

**Comment by xylar** [comment]
@lconlon, if you would review the design doc, that would be great!

@mark-petersen and @lconlon, to build the docs locally, you need to install [Miniconda3](https://docs.conda.io/en/latest/miniconda.html), add the [conda-forge](https://conda-forge.org/#about) channel, and run:
```
conda create -y -n mpas_docs python=3.8 sphinx mock sphinx_rtd_theme
conda activate mpas_docs
export DOCS_VERSION="test"
cd docs
make html
```
Then, open `_build/html/index.html` in your browser.  @caozd999 has trouble building on his Mac.  Hopefully, this has been solved but maybe use Grizzly as a backup.

**Comment by caozd999** [comment]
> @lconlon, if you would review the design doc, that would be great!
> 
> @mark-petersen and @lconlon, to build the docs locally, you need to install [Miniconda3](https://docs.conda.io/en/latest/miniconda.html), add the [conda-forge](https://conda-forge.org/#about) channel, and run:
> 
> ```
> conda create -y -n mpas_docs python=3.8 sphinx mock sphinx_rtd_theme
> conda activate mpas_docs
> export DOCS_VERSION="test"
> cd docs
> make html
> ```
> 
> Then, open `_build/html/index.html` in your browser. @caozd999 has trouble building on his Mac. Hopefully, this has been solved but maybe use Grizzly as a backup.

Thanks @xylar. After I replaced tabs by spaces and added spaces between bullet lines, I got the right document format on html.  And I found it much faster to build html locally when I turn LANL VPN off.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:165]: I follow the python specification of 4 spaces but I don't think RST/sphinx cares.

Comment (xylar) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst]: I looked into this further.  The preference is for 3 spaces: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/lists.html#unordered-list-inside-ordered-list

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:280]: Based on this link: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/lists.html#code-images-and-other-content-inside-lists

```suggestion
- *Zyserman and Fredsoe, 1994*

  .. math::

      C_{ref} = \frac{0.331(\theta - 0.045)^{1.75}}{1+0.72(\theta-0.045)^{1.75}}

  and :math:`z_{ref}=2D_{50}`. Again, :math:`\theta` is the Shields parameter and
:math:`D_{50}` is the sediment median grain diameter.

- *Goldstein et al., 2014*

  .. math::

      C_{ref} = \left[ \frac{0.328U_b}{0.0688+1000D_{50}}\right]^2

  at height :math:`z_{ref}=0.01 m`, where :math:`U_b` is the magnitude of the bottom velocity (:math:`ms^{-1}`)

```

Comment (xylar) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:280]: That is, everything but the "bullet" lines need to be indented 2 spaces.

Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:280]: Got it.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:315]: This needs to be indented, too:
```suggestion
  where :math:`\nu` is the kinematic viscosity of water (:math:`m^2s^{-1}`)
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:323]:
```suggestion
  in which :math:`D_* = D_{50}\left[\frac{(s-1)g}{\nu ^2}\right]^{1/3}` is dimensionless grain size.
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:475]: Again, a corresponding variable name from Algorithm Design: Suspended sediment transport would be helpful for each variable here.

The bottom velocity might warrant some discussion.  For one, the velocity magnitude isn't necessarily available at cell centers (but it can be computed from the kinetic energy).  For another, you may want to average over more than one layer to smooth out jumps depending on what vertical coordinates and bathymetries this capability supports (see my suggested requirement related to this.)

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:484]:
```suggestion
A logical variable need to be defined to switch ON/OFF the morphological change: :code:`config_morphological_change_enabled`.
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (xylar) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:477]: This may not seem like it but it might be the hardest change to make.  You may want to meet us with me, @dengwirda, @cbegeman and @stephenprice to discuss this because it may relate to work we are doing for ice-shelf cavities.  Currently, `bottomDepth` is the (positive-down) bathymetry in MPAS-Ocean, but it isn't allowed to change in time.  It is also intimately tied to the z* vertical coordinate that is currently the default and is implicitly assumed in at least a few places in the code that @cbegeman has discussed with us (related to the computation of vertical velocities between layers).

A change to the model to allow `bottomDepth` to change with time could be a pretty big project and could potentially be a bit disruptive.  @matthewhoffman may also be interested in this work, since it also relates to his early career project.  In sort, this might be a big enough project to need a design document all on its own with input from a lot of different people.

**Review by xylar — COMMENTED** [review]
@caozd999, this is a great document!  I've made a bunch of comments.  Probably the second to last on Implementation: Morphological change is the most critical for us to talk about.  Changing the bathymetry is likely to be a pretty big project!

**Review by lconlon — COMMENTED** [review]
I think the science here looks fine and is in line with what we've already talked about. I don't have any specific comments on it but I'm happy to chat about specifics if needed. Thanks for putting this together!

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (cbegeman) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:477]: @caozd999, @lconlon and I have had a conversation about implementing morphological change to the bathymetry. Correct me if I'm wrong, but I think our conclusion was that this development wasn't strictly necessary for ICoM's goals and that it likely wouldn't be feasible in the [current development timeframe](https://icom.atlassian.net/wiki/spaces/ICOM/pages/101711907/Overview). I'd suggest removing it from this design doc and if we find ourselves ahead of the timeline or if this development seems necessary for ICoM then we create a new design document for morphological change alone.

Comment (lconlon) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:477]: I'm trying to remember that conversation now....would erosion and deposition still work correctly if we don't change the bed elevation in the model? I remember that we decided that changing coastline morphology was out of the scope of this project, but I can't remember where we landed on bed elevation.

Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:477]: @cbegeman you are right, we don't have to implement the morphological change component for ICoM now; I put it in this doc for a more general model component introduction.  As @xylar and @cbegeman suggested, I will remove morphological change from the design doc, and we can prepare a single design doc for when necessary. But it is good to know that this would influence/interest other people if we tend to do it; seems to me a necessary model development direction.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:141]: I plan to implement both and let user choose.

**Comment by caozd999** [comment]
> I think the science here looks fine and is in line with what we've already talked about. I don't have any specific comments on it but I'm happy to chat about specifics if needed. Thanks for putting this together!

Thanks @lconlon. Within the ICoM scope, I think we'd better remove the morphological change part, but we do consider the erosion and deposition when we solve SSC. Does this sound good to you?

**Comment by lconlon** [comment]
> > I think the science here looks fine and is in line with what we've already talked about. I don't have any specific comments on it but I'm happy to chat about specifics if needed. Thanks for putting this together!
> 
> Thanks @lconlon. Within the ICoM scope, I think we'd better to remove the morphological change part, but we do consider the erosion and deposition when we solve SSC. Does this sound good to you?

That sounds good. Thanks for all your work on this!

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:477]: @lconlon we would just assume the erosion and deposition are not strong enough to significantly change either the bed layer sediment mass or the sediment layer thickness. So it would work. And yes you are right - we had a discussion before and the coastline morphology won't be in the scope either.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:43]: corrected.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:42]: corrected.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:37]: corrected.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:38]: corrected.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:363]: I think these are intermediate variables that maybe don't need to be defined externally.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:458]: Yes, that's right. I will add `config_sediment_settling_formulation`in the registry file.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:484]: corrected.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst:414]: `\rho_w` should be from the lowest ocean layer. I will remove the wave effects for just now.

**Comment by xylar** [comment]
@caozd999, how are you feeling about this one?  Is it at a stage where you're ready to merge it?  You can always edit it later.

We're in the process of merging PRs wherever we can to get ready for some pretty big changes to MPAS-Ocean.  Happy to chat about it more on Slack.

**Comment by caozd999** [comment]
> @caozd999, how are you feeling about this one? Is it at a stage where you're ready to merge it? You can always edit it later.
> 
> We're in the process of merging PRs wherever we can to get ready for some pretty big changes to MPAS-Ocean. Happy to chat about it more on Slack.

Thanks @xylar. I think this version is ready to merge, and we can edit it later if necessary.

**Comment by mark-petersen** [comment]
@xylar @lconlon @cbegeman if any of you are done reviewing, please approve. This PR is good for an 'off' day when Jon is out.

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (cbegeman) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:211]:
```suggestion
We provide two options to compute :math:`C_{ref}` and :math:`z_{ref}`:
```

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (cbegeman) [inline docs/ocean/design_docs/prognostic-sediment-transport.rst:296]:
```suggestion
- A logical variable to switch ON/OFF the bedload transport: :code:`config_sediment_bedload_enable`
```
For consistency with some of the other config variables

**Comment by cbegeman** [comment]
@caozd999 and @mark-petersen I'm happy to have this merged as-is.

**Review by cbegeman — APPROVED** [review]

**Review by xylar — APPROVED** [review]

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (mark-petersen) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst]: done

File: docs/ocean/design_docs/prognostic-sediment-transport.rst
Comment (mark-petersen) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst]: I changed all these, thanks. @caozd999 all the flags use `enable` at the end without the d

Comment (caozd999) [reply docs/ocean/design_docs/prognostic-sediment-transport.rst]: @mark-petersen Many thanks Mark.

**Pull request merged**
